src/tests/intg/files_ops.py

```diff
@@ -0,0 +1,155 @@
+#
+# SSSD integration test - operations on UNIX user and group database
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 only
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import ent
+import os.path
+import tempfile
+import pytest
+from util import backup_envvar_file, restore_envvar_file
+
+@pytest.fixture
+def passwd_ops_setup(request):
+    pwd_file = os.environ["NSS_WRAPPER_PASSWD"]
+    backup_envvar_file("NSS_WRAPPER_PASSWD")
+    request.addfinalizer(lambda: restore_envvar_file("NSS_WRAPPER_PASSWD"))
+    pwd_ops = PasswdOps(pwd_file)
+    return pwd_ops
+
+
+@pytest.fixture
+def group_ops_setup(request):
+    grp_file = os.environ["NSS_WRAPPER_GROUP"]
+    backup_envvar_file("NSS_WRAPPER_GROUP")
+    request.addfinalizer(lambda: restore_envvar_file("NSS_WRAPPER_GROUP"))
+    grp_ops = GroupOps(grp_file)
+    return grp_ops
+
+
+@pytest.fixture
+def group_db_setup(request):
+    group = request.param
+    grp_ops = group_ops_setup(request)
+    grp_ops.groupadd(**group)
+    ent.assert_group_by_name(group['name'], group)
+    return grp_ops
+
+
+class FilesOps(object):
+    """
+    A naive implementation of operations as a basis for user or group
+    operations. Uses rename to (hopefully) trigger the same fs-level
+    notifications as shadow-utils would.
+    """
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.tmp_dir = os.path.dirname(self.file_name)
+
+    def _get_named_line(self, name, contents):
```

Comment:
you might use function decorator `@staticmethod`  for methods which does not use `self`
and remove it from arguments
e.g. _get_named_line, _pwd2line, _grp2line